Composer/packages/adaptive-form/src/components/Comment.tsx

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import React, { useState, useCallback, useRef, useEffect } from 'react';
+import formatMessage from 'format-message';
+import { IconMenu } from '@bfc/ui-shared';
+import { TextField, ITextField } from 'office-ui-fabric-react/lib/TextField';
+import { IContextualMenuItem } from 'office-ui-fabric-react/lib/ContextualMenu';
+
+import { Link } from './Link';
+
+export const MAX_CHARS_FOR_SINGLE_LINE = 65;
+
+const styles = {
+  note: (isEditing = false) => css`
+    background-color: #fff4ce;
+    padding: 14px 10px 10px 10px;
+    position: relative;
+    border-radius: 3px;
+
+    display: ${isEditing ? 'none' : 'block'};
+  `,
+  menu: css`
+    position: absolute;
+    right: 4px;
+    top: 8px;
+  `,
+  noteBody: (truncate = true) => css`
+    font-size: 12px;
+    margin: 0;
+    padding-right: 18px;
+    // https://css-tricks.com/line-clampin/#weird-webkit-flexbox-way
+    ${truncate
+      ? `
+      overflow: hidden;
+      max-height: 80px;
+      display: -webkit-box;
+      -webkit-line-clamp: 5;
+      -webkit-box-orient: vertical;
+      `
+      : undefined}
+  `,
+  showMore: css`
+    display: flex;
+    justify-content: flex-end;
+    margin-top: 5px;
+  `,
+};
+
+type CommentProps = {
+  comment?: string;
+  onChange: (newComment?: string) => void;
+};
+
+const Comment: React.FC<CommentProps> = ({ comment, onChange }) => {
+  const [isEditing, setIsEditing] = useState(!comment);
+  const [isMultiline, setIsMultiline] = useState((comment ?? '').length >= MAX_CHARS_FOR_SINGLE_LINE);
+  const [showMore, setShowMore] = useState(false);
+  const [needsTruncation, setNeedsTruncation] = useState(false);
+  const noteRef = useRef<HTMLParagraphElement | null>(null);
+  const textfieldRef = useRef<ITextField>(null);
+  const editingStateRef = useRef<boolean>(isEditing);
+
+  useEffect(() => {
+    const el = noteRef.current;
+    setNeedsTruncation(Boolean(el && el.scrollHeight > el.clientHeight));
+  }, []);
+
+  useEffect(() => {
+    // send focus to the text field if toggling from view mode to edit mode
+    if (!editingStateRef.current && isEditing) {
+      textfieldRef.current?.focus();
+    } else if (!isEditing) {
+      const el = noteRef.current;
+      setNeedsTruncation(Boolean(el && el.scrollHeight > el.clientHeight));
+    }
+    editingStateRef.current = isEditing;
+  }, [isEditing]);
+
+  const handleChange = useCallback(
+    (_e, val?: string) => {
+      onChange(val);
```

Comment:
Do you need a check for undefined before you call onChange. Wonder why Fluent makes the type for value  undefined.